### PR TITLE
Add additional validation

### PR DESCRIPTION
**Pull request by wilmhit**

### CODE REVIEW

File: graph_coloring/graph_validation.py [hunk 1]
```diff
@@ -9,13 +9,24 @@ def get_error_if_invalid(graph: AdjMatrix) -> Optional[str]:
         return "The provided graph is not square"
     if not has_zeros_on_diagonal(graph):
         return "The provided graph should only have zeros on its diagonal"
+    if not edges_greater_equal_vertices(graph):
+        return "Graph must have more edges than vertices"
```
Comment (songofroland) [inline graph_coloring/graph_validation.py:13]: `The provided graph`

File: graph_coloring/graph_validation.py [hunk 2]
```diff
@@ -9,13 +9,24 @@ def get_error_if_invalid(graph: AdjMatrix) -> Optional[str]:
...
 
 def has_zeros_on_diagonal(graph: AdjMatrix) -> bool:
     return all(graph[x][x] == 0 for x in range(len(graph)))
+
+
+def edges_greater_equal_vertices(graph: AdjMatrix) -> bool:
+    vertices = len(graph)
+    edges = 0
+    for row_num in range(len(graph)):
+        for col_num in range(row_num, len(graph)):
```
Comment (songofroland) [inline graph_coloring/graph_validation.py:29]: u could easily use `list.count` here

File: tests/test_graph_validation.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import pytest
+from graph_coloring.custom_types import AdjMatrix
+from graph_coloring.graph_validation import edges_greater_equal_vertices
+
+@pytest.mark.parametrize(
+	"graph",
+	[
+		[
```
Comment (songofroland) [inline tests/test_graph_validation.py:8]: format

**Review by songofroland — APPROVED** [review]

File: tests/test_graph_validation.py (hunk 3 above)
Comment (wilmhit) [reply tests/test_graph_validation.py:8]: wtf vscode why

File: graph_coloring/graph_validation.py (hunk 2 above)
Comment (wilmhit) [reply graph_coloring/graph_validation.py:30]: Next time

**Pull request merged**